dask/base.py

```diff
@@ -172,9 +172,9 @@ def normalize_array(x):
                 data = md5(b'-'.join([str(item).encode() for item in x.flat])).hexdigest()
         else:
             try:
-                data = md5(x.view('i1').data).hexdigest()
+                data = md5(x.ravel().view('i1').data).hexdigest()
```

Comment:
The hash function runs at around 400MB/s on my laptop,  memcopy is 3000MB/s.  These seems non-trivial.  Will make a change when I get a chance.